src/codegen/generators/data_type_generator.rs

```diff
@@ -135,10 +133,11 @@ impl<'ink, 'b> DataTypeGenerator<'ink, 'b> {
                         self.index.find_type(inner_type_name).unwrap(),
                     )
                     .unwrap();
-                let array_type = self
-                    .create_nested_array_type(inner_type, dimensions.clone())
-                    .into();
-                Ok(array_type)
+
+                self.create_nested_array_type(inner_type, dimensions.clone())
+                    .map(|it| it.as_basic_type_enum())
+                    .map_err(|err| CompileError::codegen_error(err, SourceRange::undefined()))
+                //TODO error location
```

Comment:
I think location is available on the Datatype (passed as param)

Reply:
sorry no its not ... but i added an issue for it #285